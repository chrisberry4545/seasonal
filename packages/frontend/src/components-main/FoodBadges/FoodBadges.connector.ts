import { connect } from 'react-redux';
import {
  FoodBadges
} from './FoodBadges';
import {
  badgeItemClicked,
  selectCurrentFoodDetailsBadges,
  selectIsFoodDataOrBasicSeasonsLoading
} from '@chrisb-dev/seasonal-shared-frontend-redux';
import { IState } from '../../store';
import {
  IFoodBadgesInputProps,
  IFoodBadgesDispatchProps
} from './FoodBadges.interface';
import { Dispatch } from 'redux';

const mapStateToProps = (
  state: IState
): IFoodBadgesInputProps => ({
  badges: selectCurrentFoodDetailsBadges(state),
  isLoading: selectIsFoodDataOrBasicSeasonsLoading(state)
});

const mapDispatchToProps = (
  dispatch: Dispatch
): IFoodBadgesDispatchProps => ({
  onBadgeClicked: (recipeId: string) => (
    dispatch(badgeItemClicked(recipeId))
  )
});

export const FoodBadgesConnecter = connect(
  mapStateToProps,
  mapDispatchToProps
)(FoodBadges);
