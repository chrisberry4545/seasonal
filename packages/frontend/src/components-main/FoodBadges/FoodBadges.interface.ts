import { IBadge } from '@chrisb-dev/seasonal-shared-models';

export interface IFoodBadgesInputProps {
  badges: IBadge[] | undefined;
}
