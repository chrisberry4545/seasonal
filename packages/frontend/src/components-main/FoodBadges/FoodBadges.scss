@import './../../styles/settings';
@import './../../styles/tools';

.c-food-badges {
  padding: 1.5rem 1rem 2rem;
  text-align: center;
  border-bottom: 1px solid $grey-med;
}

.c-food-badges__heading {
  margin-bottom: 1rem;
  display: block;
}

.c-food-badges__list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.c-food-badges__badge {
  color: $black-color;
  padding: 0.75rem;
  border-radius: 8px;
  margin: 0 .5rem 1rem;
  box-shadow: 1px 2px 4px 1px transparentize($black-color, .75);
  position: relative;
  background-color: $alt-color-1;
}
