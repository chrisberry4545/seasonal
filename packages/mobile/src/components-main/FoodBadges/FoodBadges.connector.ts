import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import {
  FoodBadges
} from './FoodBadges';
import {
  selectCurrentFoodDetailsBadges,
  badgeItemClicked,
  selectIsFoodDataOrBasicSeasonsLoading
} from '@chrisb-dev/seasonal-shared-frontend-redux';
import { IState } from '../../store';
import {
  IFoodBadgesInputProps,
  IFoodBadgesDispatchProps
} from './FoodBadges.interface';

const mapStateToProps = (
  state: IState
): IFoodBadgesInputProps => ({
  badges: selectCurrentFoodDetailsBadges(state),
  isLoading: selectIsFoodDataOrBasicSeasonsLoading(state)
});

const mapDispatchToProps = (
  dispatch: Dispatch
): IFoodBadgesDispatchProps => ({
  onBadgeClicked: (recipeId: string) => (
    dispatch(badgeItemClicked(recipeId))
  )
});

export const FoodBadgesConnecter = connect(
  mapStateToProps,
  mapDispatchToProps
)(FoodBadges);
