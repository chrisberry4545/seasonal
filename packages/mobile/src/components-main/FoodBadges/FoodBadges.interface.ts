import { IBadge } from '@chrisb-dev/seasonal-shared-models';

export interface IFoodBadgesInputProps {
  badges: IBadge[] | undefined;
}

export interface  IFoodBadgesDispatchProps {
  onBadgeClicked: (badgeId: string) => void;
}

export interface IFoodBadgesProps
  extends IFoodBadgesInputProps, IFoodBadgesDispatchProps {}
