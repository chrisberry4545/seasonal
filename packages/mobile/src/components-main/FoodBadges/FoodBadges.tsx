import React, { FC } from 'react';
import {
  IFoodBadgesInputProps
} from './FoodBadges.interface';
import { TextHeadingSmall, TextMedium } from '../../components-elements';
import { ViewStyle, View, TextStyle, StyleSheet } from 'react-native';
import { colors } from '../../styles/colors';
import { styles } from '../../styles';
import { shadows } from '../../styles/shadows';

const styleBadges: ViewStyle = {
  borderBottomWidth: StyleSheet.hairlineWidth,
  borderColor: styles.colors.greyMed,
  paddingBottom: 33
};

const styleBadgesHeading: TextStyle = {
  marginBottom: 16,
  marginTop: 24,
  textAlign: 'center',
  width: '100%'
};

const styleBadgesList: ViewStyle = {
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'center'
};

const styleBadgesBadgeItem: TextStyle = {
  ...shadows.level1,
  backgroundColor: colors.selectionColorAlt,
  borderRadius: 8,
  color: colors.black,
  marginBottom: 10,
  marginEnd: 10,
  padding: 10
};

export const FoodBadges: FC<IFoodBadgesInputProps> = ({
  badges
}) => (
  (badges && badges.length > 0)
    ? (
      <View style={styleBadges}>
        <TextHeadingSmall style={styleBadgesHeading}>
          Nutrition
        </TextHeadingSmall>
        <View style={styleBadgesList}>
          {
            badges.map((badge) =>
              <View
                style={styleBadgesBadgeItem}
                key={badge.id}>
                  <TextMedium>
                    {badge.name}
                  </TextMedium>
              </View>
            )
          }
        </View>
      </View>
    ) : null
);
