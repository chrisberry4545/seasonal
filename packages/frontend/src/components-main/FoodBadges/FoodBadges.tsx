import React, { FC } from 'react';

import './FoodBadges.scss';

import {
  IFoodBadgesProps
} from './FoodBadges.interface';
import { TextHeadingSmall, BareButton } from '@chrisb-dev/seasonal-shared-frontend-components';

export const FoodBadges: FC<IFoodBadgesProps> = ({
  badges,
  onBadgeClicked,
  isLoading
}) => (
  (!isLoading && badges && badges.length > 0)
    ? (
      <div className='c-food-badges'>
        <TextHeadingSmall className='c-food-badges__heading'>
          Nutrition
        </TextHeadingSmall>
        <div className='c-food-badges__list'>
          {
            badges.map((badge) =>
              <BareButton
                onClick={() => onBadgeClicked(badge.id)}
                className='c-food-badges__badge'
                key={badge.id}>
                {badge.name}
              </BareButton>
            )
          }
        </div>
      </div>
    ) : null
);
