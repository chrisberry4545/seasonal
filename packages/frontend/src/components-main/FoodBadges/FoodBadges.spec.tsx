
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { FoodBadges } from './FoodBadges';
import { IBadge } from '@chrisb-dev/seasonal-shared-models';

jest.mock('@chrisb-dev/seasonal-shared-frontend-components', () => ({
  TextHeadingSmall: () => 'TextHeadingSmall'
}));

describe('<FoodBadges />', () => {
  let wrapper: ShallowWrapper;
  const badges = [{
    color: '#000000',
    id: '1',
    name: 'Iron'
  }, {
    color: '#FFFFFF',
    id: '2',
    name: 'Vitamin A'
  }] as IBadge[];

  describe('when there are food badges', () => {
    beforeEach(() =>
      wrapper = shallow(
        <FoodBadges
          badges={badges} />
      )
    );

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());
  });

  describe('when there are no badges', () => {
    beforeEach(() =>
      wrapper = shallow(
        <FoodBadges
          badges={undefined} />
      )
    );

    test('does not render anything', () =>
      expect(wrapper.children().exists()).toBe(false));
  });

  describe('when the badges have length 0', () => {
    beforeEach(() =>
      wrapper = shallow(
        <FoodBadges
          badges={[]} />
      )
    );

    test('does not render anything', () =>
      expect(wrapper.children().exists()).toBe(false));
  });

});
