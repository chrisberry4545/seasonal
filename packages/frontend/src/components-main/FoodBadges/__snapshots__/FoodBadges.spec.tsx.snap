// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodBadges /> when there are food badges renders correctly 1`] = `
<div
  className="c-food-badges"
>
  <TextHeadingSmall
    className="c-food-badges__heading"
  >
    Nutrition
  </TextHeadingSmall>
  <div
    className="c-food-badges__list"
  >
    <BareButton
      className="c-food-badges__badge"
      key="1"
      onClick={[Function]}
    >
      Iron
    </BareButton>
    <BareButton
      className="c-food-badges__badge"
      key="2"
      onClick={[Function]}
    >
      Vitamin A
    </BareButton>
  </div>
</div>
`;
